### PR TITLE
Add edge case scenarios for Apply workflow testing

### PR DESCRIPTION
**Pull request by aoysimmons-GH**
Audit and create a plan for priority E2E tests for the Simpler Apply workflow, focusing on edge cases and user interactions.

## Summary

Fixes / Work for #7847

### CODE REVIEW

File: frontend/tests/e2e/apply/features/edge_case_apply [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  Scenario: Apply shell partially loads due to slow network
+    Given I select "Start Application"
+    When the Apply shell loads partially or times out
+    Then I should see an error message or retry option
+    And the system should not allow submission until fully loaded
+    
+  ###########################################################################
+  # Edge Case 2: User navigates too quickly
+  ###########################################################################
+  Scenario: Rapid navigation between section
```
Comment (doug-s-nava) [inline frontend/tests/e2e/apply/features/edge_case_apply]: this doesn't seem relevant as the left hand nav links only move around the page, no page -> page navigation or api calls

File: frontend/tests/e2e/apply/features/edge_case_apply [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    When I enter special characters, HTML, or very long strings
+    Then the system should validate and show appropriate errors
+    And prevent incorrect submission
+    
+  ###########################################################################
+   # Edge Case 4: Page refresh at critical moments
+  ###########################################################################
+  Scenario: Refreshing page during submission
+    Given I have completed all required fields
+    When I refresh the page while submitting
```
Comment (doug-s-nava) [inline frontend/tests/e2e/apply/features/edge_case_apply]: let's be clear about "submission" which happens for the full application and "saving" which happens for an individual form. Is this talking about the form or the application? Also, please expand on "while submitting" - what are the steps to get into a "submitting" state here?

File: frontend/tests/e2e/apply/features/edge_case_apply (hunk 1 above)
Comment (aoysimmons-GH) [reply frontend/tests/e2e/apply/features/edge_case_apply]: That's is make sense. I am removing this scenario.

File: frontend/tests/e2e/apply/features/edge_case_apply [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+   # Edge Case 3: Page refresh at critical moments at save
+  ###########################################################################
+  Scenario: Refreshing page during save
+    Given I have completed all required fields
+    And I click to save
+    When I refresh the page while saving
+    Then my entered data should persist
+
+  ###########################################################################
+  # Edge Case 4A: Page refresh resumes submission (API-aware)
```
Comment (doug-s-nava) [inline frontend/tests/e2e/apply/features/edge_case_apply]: this isn't how our system works. Maybe take 4a and 4b out for now so we can move forward, and me, Chris K or someone from the apply team can walk you through how submission works and we can revisit
